### PR TITLE
refactor: append return statements after response callback

### PR DESCRIPTION
**Pull request by kylerwsm**
## Problem

Some of the response statements are not followed by a return statement.

## Solution

- Append return statements to the next line of response statements.
- Also added explicit status code to the existing `res.json` statement.
- Disabled an Eslint rule that removes return statements which it deems unnecessary.

### CODE REVIEW

**Comment by JasonChong96** [comment]
Technically in those cases the function call ends after the response statements. But I think its an improvement to have the explicit return statements to prevent bugs in case anyone adds any lines after.

**Comment by kylerwsm** [comment]
@JasonChong96 I agree. Additionally, this issue might had been exacerbated by eslint, which right now removes return statements it deems unnecessary. This creates problems as we need to make conscious efforts to remember to add the return statements back when it becomes less unnecessary. I talked to @liangyuanruo about it, and am working on removing that eslint check.

**Comment by yong-jie** [comment]
fired gitpod and did a grep project-wide grep check. confirmed that all relevant `res.status and res.json, res.ok, res.badRequest` etc calls now have the `return` statement

File: src/server/controllers/LoginController.ts [hunk 1]
```diff
@@ -60,18 +64,22 @@ export class LoginController implements LoginControllerInterface {
...
+        return
+      }
+      if (error instanceof NotFoundError) {
         res.unauthorized(jsonMessage('OTP expired/not found.'))
-      } else {
-        res.serverError(jsonMessage(error.message))
+        return
       }
+      res.serverError(jsonMessage(error.message))
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/LoginController.ts:82]: I think we should retain the `else-if` and `else` blocks here for greater code clarity.

File: src/server/controllers/LoginController.ts [hunk 2]
```diff
@@ -84,9 +92,10 @@ export class LoginController implements LoginControllerInterface {
     if (user) {
       const response = { ...jsonMessage('Logged in'), user }
       res.ok(response)
-    } else {
-      res.notFound(jsonMessage('User session not found'))
+      return
     }
+    res.notFound(jsonMessage('User session not found'))
+    return
```
Comment (liangyuanruo) [inline src/server/controllers/LoginController.ts:98]: similarly here

File: src/server/controllers/RedirectController.ts [hunk 3]
```diff
@@ -81,9 +81,10 @@ export class RedirectController implements RedirectControllerInterface {
           gaOnLoad: EventAction.LOADED,
           gaOnProceed: EventAction.PROCEEDED,
         })
-      } else {
-        res.status(302).redirect(longUrl)
+        return
       }
+      res.status(302).redirect(longUrl)
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/RedirectController.ts:87]: and here

File: src/server/controllers/UserController.ts [hunk 4]
```diff
@@ -44,15 +44,19 @@ export class UserController implements UserControllerInterface {
...
+      if (error instanceof AlreadyExistsError) {
         res.badRequest(jsonMessage(error.message, MessageType.ShortUrlError))
-      } else {
-        logger.error(`Error creating short URL:\t${error}`)
-        res.badRequest(jsonMessage('Server error.'))
+        return
       }
+      logger.error(`Error creating short URL:\t${error}`)
+      res.badRequest(jsonMessage('Server error.'))
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/UserController.ts:59]: and here

File: src/server/controllers/UserController.ts [hunk 5]
```diff
@@ -80,13 +84,15 @@ export class UserController implements UserControllerInterface {
...
       if (error instanceof NotFoundError) {
         res.notFound(jsonMessage(error.message))
-      } else {
-        logger.error(`Error editing URL:\t${error}`)
-        res.badRequest(jsonMessage(`Unable to edit short link "${shortUrl}"`))
+        return
       }
+      logger.error(`Error editing URL:\t${error}`)
+      res.badRequest(jsonMessage(`Unable to edit short link "${shortUrl}"`))
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/UserController.ts:95]: and here

File: src/server/controllers/UserController.ts [hunk 6]
```diff
@@ -107,15 +113,19 @@ export class UserController implements UserControllerInterface {
...
+      if (error instanceof AlreadyOwnLinkError) {
         res.badRequest(jsonMessage(error.message))
-      } else {
-        logger.error(`Error transferring ownership of short URL:\t${error}`)
-        res.badRequest(jsonMessage('An error has occured'))
+        return
       }
+      logger.error(`Error transferring ownership of short URL:\t${error}`)
+      res.badRequest(jsonMessage('An error has occured'))
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/UserController.ts:128]: and here

File: src/server/controllers/UserController.ts [hunk 7]
```diff
@@ -151,12 +161,14 @@ export class UserController implements UserControllerInterface {
...
     } catch (error) {
       if (error instanceof NotFoundError) {
         res.notFound(error.message)
-      } else {
-        res.serverError(jsonMessage('Error retrieving URLs for user'))
+        return
       }
+      res.serverError(jsonMessage('Error retrieving URLs for user'))
+      return
```
Comment (liangyuanruo) [inline src/server/controllers/UserController.ts:171]: and here

**Comment by kylerwsm** [comment]
Discussed with @liangyuanruo that the removal of else and else-if statements is because of an [ESLint](https://eslint.org/docs/rules/no-else-return) rule, and we will be keeping the rule for now.

**Pull request merged**